core/src/test/scala/codesearch/core/search/SnippetsGrouperSpec.scala

```diff
@@ -0,0 +1,90 @@
+package codesearch.core.search
+
+import cats.data.NonEmptyList
+import codesearch.core.config.SnippetConfig
+import codesearch.core.search.SnippetsGrouper.SnippetInfo
+import org.scalatest.{Matchers, WordSpec}
+
+class SnippetsGrouperSpec extends WordSpec with Matchers {
```

Comment:
Yay, tests, thanks!

Not sure this test will work properly, though – it might be better to have a folder with some fake Haskell code in the repo and use it for tests. But it's fine to do it in a separate PR.